### PR TITLE
Fix #426, change history path of ptpython

### LINKED ISSUE #426: ptpython  now use ~/.local/share/ptpython/history     instead of      ~/.ptpython/history
should  this linein  pudb/shell.py be changed? 
` history_filename = os.path.expanduser("~/.local/share/ptpython/history") `

### PR DESCRIPTION
**Pull request by Freed-Wu**
I have tested it can work.

### CODE REVIEW

File: pudb/shell.py [hunk 1]
```diff
@@ -249,7 +249,7 @@ class DummyMod:
 def run_ptpython_shell(globals, locals):
     # Use the default ptpython history
     import os
-    history_filename = os.path.expanduser("~/.ptpython/history")
```
Comment (mvanderkamp) [inline pudb/shell.py:252]: This is still supported as the "legacy history file":
https://github.com/prompt-toolkit/ptpython/blob/d8b5c90d20cc1e8d88837aa7163289950c231f74/ptpython/entry_points/run_ptpython.py#L131

Ideally, we would look up the history file using ptpython's methods, or at least emulate them or allow users to configure it to the desired location.

Comment (Freed-Wu) [reply pudb/shell.py:252]: How do you think of `os.access(os.path.expanduser("~/.ptpython/history"), os.W_OK)`?

Comment (Freed-Wu) [reply pudb/shell.py:252]: If `~/.ptpython/history` don't exist, let us use `~/.local/share/ptpython/history`.

Comment (inducer) [reply pudb/shell.py:252]: Why not just call `ptpython.get_config_and_history_file`, as @mvanderkamp suggests?

https://github.com/prompt-toolkit/ptpython/blob/d8b5c90d20cc1e8d88837aa7163289950c231f74/ptpython/entry_points/run_ptpython.py#L131

Comment (inducer) [reply pudb/shell.py:252]: Thanks! Failed the linter, otherwise LGTM!

**Comment by Freed-Wu** [comment]
I have remove unused `import os`. Thanks!

**Comment by inducer** [comment]
LGTM, thanks!

**Comment by inducer** [comment]
Replaced by #495.

**Pull request closed without merging**